crypto/src/main/java/net/consensys/cava/crypto/mikuli/group/G2Point.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.cava.crypto.mikuli.group;
+
+import org.apache.milagro.amcl.BLS381.BIG;
+import org.apache.milagro.amcl.BLS381.ECP2;
+
+public final class G2Point implements Group<G2Point> {
+  private final ECP2 point;
+  private static final int fpPointSize = BIG.MODBYTES;
+
+  public G2Point(ECP2 point) {
+    if (point == null) {
```

Comment:
same trick with package-info.java in that other package.

Reply:
Actually, if it's part of the module API (e.g. a public method in a public class), then you should still check arguments. However, prefer `Objects.requireNonNull(point)` for these checks.